envoimoinscher.php

```diff
@@ -995,69 +1017,129 @@ public function ordersTable()
 
-		$pager_error = new Pager(array(
-			'url' => 'index.php?controller=AdminEnvoiMoinsCher&pe='.$page_emc.'&po='.$page_others,
-			'tag' => 'pr',
-			'before' => 5,
-			'after' => 5,
-			'all' => $orders_count['errors'],
-			'page' => $page_error,
-			'perPage' => $per_page
-		));
-		$start = ($page_error - 1) * $per_page;
-		$limits['errors'] = 'LIMIT '.$start.','.$per_page;
-		$smarty->assign('pager_error', $pager_error->setPages());
+		$orders = $this->model->getEligibleOrders($params, $limits);
 
-		$orders = $this->model->getEligibleOrders(array('lang' => $cookie->id_lang), $limits);
+		//get enabled carriers
+		$sql = 'SELECT id_carrier, name FROM '._DB_PREFIX_.'carrier WHERE deleted=0';
+		$enabled_carriers = Db::getInstance()->ExecuteS($sql);
+
+		//send filter settings back
+		if (isset($params['filterBy']))
+		{
+			$smarty->assign('filters', $params['filterBy']);
+			$filter_url = '';
+			if (isset($params['filterBy']['type_order']))
+				$filter_url .= '&type_order='.$params['filterBy']['type_order'];
+			if (isset($params['filterBy']['filter_id_order']))
+				$filter_url .= '&filter_id_order='.$params['filterBy']['filter_id_order'];
+			if (isset($params['filterBy']['status']) && is_array($params['filterBy']['status']))
+				foreach ($params['filterBy']['status'] as $key => $value)
+					$filter_url .= '&status[]='.$value;
+			if (isset($params['filterBy']['carriers']))
+				$filter_url .= '&carriers='.$params['filterBy']['carriers'];
+			if (isset($params['filterBy']['start_order_date']))
+				$filter_url .= '&start_order_date='.$params['filterBy']['start_order_date'];
+			if (isset($params['filterBy']['end_order_date']))
+				$filter_url .= '&end_order_date='.$params['filterBy']['end_order_date'];
+			if (isset($params['filterBy']['recipient']))
+			{
+				$filter_url .= '&recipient=';
+				$i = 0;
+				foreach ($params['filterBy']['recipient'] as $key => $value)
+				{
+					if ($i == 0)
+						$filter_url .= $value;
+					else
+						$filter_url .= '+'.$value;
+					$i++;
+				}
+			}
+			$smarty->assign('filterUrl', $filter_url);
+		}
+		//else send default filter settings
+		else
+		{
+			$params['filterBy']['type_order'] = $config['EMC_FILTER_TYPE_ORDER'];
+			$params['filterBy']['status'] = explode(';', $config['EMC_FILTER_STATUS']);
+			$params['filterBy']['carriers'] = $config['EMC_FILTER_CARRIERS'];
+			switch ($config['EMC_FILTER_START_DATE'])
+			{
+				case 'all':
+					break;
+
+				case 'year':
+					$params['filterBy']['start_order_date'] = date('Y-m-d', strtotime('-1 year'));
+					break;
+
+				case 'month':
+					$params['filterBy']['start_order_date'] = date('Y-m-d', strtotime('-1 month'));
+					break;
+
+				case 'week':
+					$params['filterBy']['start_order_date'] = date('Y-m-d', strtotime('-1 week'));
+					break;
+
+				case 'day':
+					$params['filterBy']['start_order_date'] = date('Y-m-d', strtotime('-1 day'));
+					break;
+			}
+			$smarty->assign('filters', $params['filterBy']);
+		}
```

Comment:
Done with the incoming revision for $filter_url. For the $sql part, even if your solution is faster, i think it is easier to understand the code if the $sql "filter part" is built in getEligibleOrders function